src/main/java/slimeknights/tconstruct/gadgets/item/slimesling/MagmaSlimeSlingItem.java

```diff
@@ -0,0 +1,63 @@
+package slimeknights.tconstruct.gadgets.item.slimesling;
+
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.entity.player.ServerPlayerEntity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.math.AxisAlignedBB;
+import net.minecraft.util.math.vector.Vector3d;
+import net.minecraft.world.World;
+import slimeknights.tconstruct.common.Sounds;
+import slimeknights.tconstruct.library.network.TinkerNetwork;
+import slimeknights.tconstruct.tools.common.network.EntityMovementChangePacket;
+
+import java.util.List;
+
+public class MagmaSlimeSlingItem extends BaseSlimeSlingItem {
+
+  public MagmaSlimeSlingItem(Properties props) {
+    super(props);
+  }
+
+  /** Called when the player stops using an Item (stops holding the right mouse button). */
+  @Override
+  public void onPlayerStoppedUsing(ItemStack stack, World worldIn, LivingEntity entityLiving, int timeLeft) {
+    if (!(entityLiving instanceof PlayerEntity)) {
+      return;
+    }
+
+    PlayerEntity player = (PlayerEntity) entityLiving;
+    float f = getForce(stack, timeLeft);
+
+    float range = 5F;
+    Vector3d start = player.getEyePosition(1F);
+    Vector3d look = player.getLookVec();
+    Vector3d direction = start.add(look.x * range, look.y * range, look.z * range);
+    AxisAlignedBB bb = player.getBoundingBox().expand(look.x * range, look.y * range, look.z * range).expand(1, 1, 1);
+    List<Entity> entitiesInArea = player.getEntityWorld().getEntitiesWithinAABBExcludingEntity(player, bb);
+    double dist = range;
+    Entity closestEntity = null;
+    for (Entity entity : entitiesInArea) {
+      if (entity.getBoundingBox().intersects(Math.min(start.x, direction.x), Math.min(start.y, direction.y), Math.min(start.z, direction.z), Math.max(start.x, direction.x), Math.max(start.y, direction.y), Math.max(start.z, direction.z))) {
```

Comment:
These min and max values are not changing with each iteration. It may be a micro-optimization, but we might as well calculate them just once before the for loop. Can give them names like `minX`, `maxY`, etc.